pkg/cmd/taskrun/list.go

```diff
@@ -0,0 +1,147 @@
+
+package taskrun
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"text/tabwriter"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/formatted"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	cliopts "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+const (
+	msgNoTRsFound = "No taskruns found."
+)
+
+func listCommand(p cli.Params) *cobra.Command {
+	f := cliopts.NewPrintFlags("list")
+	eg := `
+# List all TaskRuns of Task name 'foo'
+tkn taskrun list  foo -n bar
+
+# List all taskruns in a namespaces 'foo'
+tkn pr list -n foo \n",
+`
+
+	c := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List the taskruns",
+		Example: eg,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var task string
+
+			if len(args) > 0 {
+				task = args[0]
+			}
+
+			trs, err := list(p, task)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to list taskruns from %s namespace \n", p.Namespace())
+				return err
+			}
+
+			output, err := cmd.LocalFlags().GetString("output")
+			if err != nil {
+				fmt.Fprint(os.Stderr, "Error: output option not set properly \n")
+				return err
+			}
+
+			if output != "" {
+				return printObject(cmd.OutOrStdout(), trs, f)
+			}
+
+			err = printFormatted(cmd.OutOrStdout(), trs, p.Time())
+			if err != nil {
+				fmt.Fprint(os.Stderr, "Failed to print Taskruns \n")
+				return err
+			}
+			return nil
+		},
+	}
+
+	f.AddFlags(c)
+
+	return c
+}
+
+func list(p cli.Params, task string) (*v1alpha1.TaskRunList, error) {
+	cs, err := p.Clientset()
+	if err != nil {
+		return nil, err
+	}
+
+	options := v1.ListOptions{}
+	if task != "" {
+		options = v1.ListOptions{
+			LabelSelector: fmt.Sprintf("tekton.dev/task=%s", task),
+		}
+	}
+
+	trc := cs.TektonV1alpha1().TaskRuns(p.Namespace())
+	trs, err := trc.List(options)
+	if err != nil {
+		return nil, err
+	}
+
+	// NOTE: this is required for -o json|yaml to work properly since
+	// tektoncd go client fails to set these; probably a bug
+	trs.GetObjectKind().SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Version: "tekton.dev/v1alpha1",
+			Kind:    "TaskRunList",
+		})
+
+	return trs, nil
+}
+
+func printObject(out io.Writer, trs *v1alpha1.TaskRunList, f *cliopts.PrintFlags) error {
+	//NOTE: no null checks for trs; caller needs to ensure it is not null
+	printer, err := f.ToPrinter()
+	if err != nil {
+		return err
+	}
+	return printer.PrintObj(trs, out)
+}
+
+func printFormatted(out io.Writer, trs *v1alpha1.TaskRunList, c clockwork.Clock) error {
+	//NOTE: no null checks for trs; caller needs to ensure it is not null
+	if len(trs.Items) == 0 {
+		fmt.Fprintln(out, msgNoTRsFound)
+		return nil
+	}
+
+	w := tabwriter.NewWriter(out, 0, 5, 3, ' ', tabwriter.TabIndent)
+	defer w.Flush()
+	fmt.Fprintln(w, "NAME\tSTARTED\tDURATION\tSTATUS\t")
```

Comment:
may want to use the same convention as `header, body, blank`

Reply:
actually isn't it the exact same as `pipelinerun` here ? :sweat_smile: